subprojects/android-test/instrumentation/src/main/kotlin/com/avito/android/instrumentation/ActivityProviderImpl.kt

```diff
@@ -0,0 +1,54 @@
+package com.avito.android.instrumentation
+
+import android.app.Activity
+import android.app.Instrumentation
+import android.content.ActivityNotFoundException
+import android.os.Looper
+import androidx.test.runner.lifecycle.ActivityLifecycleMonitor
+import androidx.test.runner.lifecycle.Stage
+import com.avito.android.Result
+
+internal class ActivityProviderImpl(
+    private val instrumentation: Lazy<Instrumentation>,
+    private val activityLifecycleMonitor: Lazy<ActivityLifecycleMonitor>
+) : ActivityProvider {
+
+    override fun getCurrentActivity(): Result<Activity> {
+        var result: Result<Activity> = notFoundResult()
+
+        if (isMainThread()) {
+            result = findResumedActivity()
+        } else {
+            try {
+                instrumentation.value.runOnMainSync {
```

Comment:
This function could lead to dead lock so it will be better if we add a timeout and if we don't get an Activity in 10 second then fail or return null